cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp_impl.hpp

```diff
@@ -93,24 +97,27 @@ std::int64_t kernels_fp<Float>::get_part_count_for_partial_centroids(sycl::queue
 
 template <typename Float>
 sycl::event kernels_fp<Float>::select(sycl::queue& queue,
-                                      const pr::ndview<Float, 2>& data,
+                                      const pr::ndview<Float, 2>& distances,
+                                      const pr::ndview<Float, 1>& centroid_squares,
                                       pr::ndview<Float, 2>& selection,
                                       pr::ndview<std::int32_t, 2>& indices,
                                       const bk::event_vector& deps) {
-    ONEDAL_ASSERT(indices.get_dimension(0) == data.get_dimension(0));
+    ONEDAL_ASSERT(indices.get_dimension(0) == distances.get_dimension(0));
     ONEDAL_ASSERT(indices.get_dimension(1) == 1);
-    ONEDAL_ASSERT(selection.get_dimension(0) == data.get_dimension(0));
+    ONEDAL_ASSERT(selection.get_dimension(0) == distances.get_dimension(0));
     ONEDAL_ASSERT(selection.get_dimension(1) == 1);
+    ONEDAL_ASSERT(centroid_squares.get_dimension(0) == distances.get_dimension(1));
 
-    const std::int64_t col_count = data.get_dimension(1);
-    const std::int64_t row_count = data.get_dimension(0);
-    const std::int64_t stride = data.get_dimension(1);
+    const std::int64_t cluster_count = distances.get_dimension(1);
+    const std::int64_t row_count = distances.get_dimension(0);
+    const std::int64_t stride = distances.get_dimension(1);
 
     const std::int64_t preffered_wg_size = 128;
```

Comment:
Please declare it as a member of kernels_fp struct, to have all config consts in one place.
Definition it here doesn't differ from `magic number` like  bk::get_scaled_wg_size_per_row(queue, cluster_count, 128);

Reply:
Fixed